### PR TITLE
Fix port already in use when running inference

### PR DESCRIPTION
**Pull request by 7174Andy**
### Description
This PR solves the port bug mentioned in the Issue https://github.com/talmolab/sleap/issues/2056. 

The cause of the bug is that the port after checking if it is free becomes TIME_WAIT state where it remains bound after being recently closed. When binding to the TIME_STATE ports, there will be an port already in use error. To solve this issue, I added a wait for 0.1 seconds in order to wait until the port is out of TIME_WAIT state after the port is bound to check if it is free to bind because I cannot directly add the `SO_REUSEADDR` that binds to other addresses without conflicts to the socket options ([Source](https://hea-www.harvard.edu/~fine/Tech/addrinuse.html)). 

### Types of changes

- [x] Bugfix
- [ ] New feature
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
https://github.com/talmolab/sleap/issues/2056

### Outside contributors checklist

- [x] Review the [guidelines for contributing](https://github.com/talmolab/sleap/blob/develop/docs/CONTRIBUTING.md) to this repository
- [x] Read and sign the [CLA](https://github.com/talmolab/sleap/blob/develop/sleap-cla.pdf) and add yourself to the [authors list](https://github.com/talmolab/sleap/blob/develop/AUTHORS)
- [x] Make sure you are making a pull request against the **develop** branch (not *main*). Also you should start *your branch* off *develop*
- [x] Add tests that prove your fix is effective or that your feature works
- [x] Add necessary documentation (if appropriate)

#### Thank you for contributing to SLEAP!
:heart:

## Summary by CodeRabbit

- **New Features**
	- Enhanced interactivity and responsiveness of the GUI for real-time updates during training.
	- Improved handling of log scale and outlier settings in the LossViewer and LossPlot.
	- Introduced a new function to find a free port for binding, enhancing port management.

- **Bug Fixes**
	- Corrected y-axis scaling for log transformations to prevent issues with values close to zero.

- **Refactor**
	- Streamlined logic for updating the plot based on training data and improved efficiency of the plotting process.
	- Updated method signatures for log scale and ignore outliers in the LossViewer class.

### CODE REVIEW

**Review by roomrys — CHANGES_REQUESTED** [review]
If this is a race condition issue, and it can this tie-up can last several minutes
>TIME_WAIT is the state that typically ties up the port for several minutes after the process has completed.

, then an automated test that quickly makes back-to-back calls to the code we use for finding a free port should be able to consistently catch this issue.

I know that the code for `find_free_port` is a nested function at the moment, but perhaps it is worth breaking out the logic inside `_setup_zmq` to a more test-able file-level function.

File: sleap/gui/utils.py [hunk 1]
```diff
@@ -12,6 +13,7 @@ def is_port_free(port: int, zmq_context: Optional[zmq.Context] = None) -> bool:
     try:
         socket.bind(address)
         socket.unbind(address)
+        time.sleep(0.1)
```
Comment (7174Andy) [inline sleap/gui/utils.py:16]: Removing `time.wait(0.1)` in this line causes an error while testing `monitor.py` consistently.

**Review by eberrigan — APPROVED** [review]

**Comment by eberrigan** [comment]
@roomrys @7174Andy were the changes made? I actually can't merge this
